### PR TITLE
Add country field to PersonForm; default to USA for patients with null country

### LINKED ISSUE #2837: Add international fields in patient address section of People Profile
As a testing site, I want the ability to test patients whose primary address is outside of the United States. Today, I cannot do this because SimpleReport does not include country or country code fields.

To amend this, we need to update the people profile page to include fields required to record international addresses. Because the majority of patients will reside in the United States, we should make USA the default if possible. This should be the same on the patient self-reg page as well.

Things to consider:

- Smarty Streets - is there anything we need to relax w/ their validation?

- ReportStream - will anything need to change re: mapping for other countries?

### PR DESCRIPTION
**Pull request by nickclyde**
## Related Issue or Background Info

- Repeat of #2847 
- Resolves #2837

## Changes Proposed

- Same as #2847, which got reverted, but also defaults the country to "USA" when editing an existing patient which has country as `null`

## Additional Information

- decisions that were made
- discussion of tradeoffs / future work
- complaints about how bad the code is

### UI
- [ ] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Review by emmastephenson — APPROVED** [review]
LGTM, thanks for fixing this Nick!

One thing I want to check on; did you verify that this works with the "save and start test" functionality?

**Comment by nickclyde** [comment]
I believe @zdeveloper was working on a fix for the save and start test issue? If that's the case, I think we'll need to wait for that to get merged before merging this PR?

File: frontend/src/index.tsx
Comment (nickclyde) [inline frontend/src/index.tsx:74]: This prevents an infinite loop in local when your Okta token expires

**Comment by nickclyde** [comment]
Unfortunately something about this PR seems to break "Save and start test", just confirmed on deploy to dev. Will look into it.

**Comment by nickclyde** [comment]
I just deployed to dev again after merging `main` and "Save and start test" seems to work. So think we are actually good to merge, I'll keep an eye out in case the issue pops up again.

**Pull request merged**
